extensions/kraken/exchange.js

```diff
@@ -0,0 +1,278 @@
+var KrakenClient = require('kraken-api'),
+  path = require('path'),
+  moment = require('moment'),
+  n = require('numbro'),
+  colors = require('colors')
+
+module.exports = function container(get, set, clear) {
+  var c = get('conf');
+
+  var public_client, authed_client;
+  var recoverableErrors = new RegExp(/(ESOCKETTIMEDOUT|ETIMEDOUT|ECONNRESET|ECONNREFUSED|ENOTFOUND|API:Invalid nonce|API:Rate limit exceeded)/);
+
+  function publicClient() {
+    if (!public_client) {
+      public_client = new KrakenClient();
+    }
+    return public_client;
+  }
+
+  function authedClient() {
+    if (!authed_client) {
+      if (!c.kraken || !c.kraken.key || c.kraken.key === 'YOUR-API-KEY') {
+        throw new Error('please configure your Kraken credentials in conf.js')
+      }
+      authed_client = new KrakenClient(c.kraken.key, c.kraken.secret);
+    }
+    return authed_client;
+  }
+
+  function joinProduct(product_id) {
+    return product_id.split('-')[0] + product_id.split('-')[1];
+  }
+
+  function retry(method, args, error) {
+    if (error.message.match(/API:Rate limit exceeded/)) {
+      var timeout = 10000;
+    } else {
+      var timeout = 2500;
+    }
+
+    console.error(('\nKraken API error - unable to call ' + method + ' (' + error + '), retrying in ' + timeout / 1000 + 's').red);
+    setTimeout(function () {
+      exchange[method].apply(exchange, args)
+    }, timeout);
+  }
+
+  var orders = {}
+
+  var exchange = {
+    name: 'kraken',
+    historyScan: 'forward',
+    makerFee: 0.16,
```

Comment:
I'm wondering about adding logic around this so that this would go down `0.02%` per tier (as per https://www.kraken.com/help/fees)
but it seems https://github.com/carlos8f/zenbot/blob/2444a9e6c77eb78f2d640d41efbfd23522116e7f/lib/engine.js#L466 only knows about the amount, and adding in tiers might be out of scope for this initial stab. Something to consider, though :)